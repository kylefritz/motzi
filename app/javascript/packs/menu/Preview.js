import React from "react";

import BakersNote from "./BakersNote";
import Deadline from "./Deadline";
import Items from "./Items";
import Subscription from "./Subscription";

export default function Preview({ user, menu }) {
  const { name, menuNote, items, deadlineDay } = menu;

  return (
    <>
      {user && <Subscription {...{ user, deadlineDay }} />}

      <h2 id="menu-name">{name}</h2>
      <Deadline menu={menu} />
      <BakersNote note={menuNote} />

      <h5>Items</h5>
      <Items marketplace items={items} />
    </>
  );
}
