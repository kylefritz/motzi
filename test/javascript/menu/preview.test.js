import React from 'react';
require('../configure_enzyme');
import { shallow } from 'enzyme';

import Preview from 'menu/Preview'
import orderData from './order-data'

test('Preview snapshot', () => {
  const wrapper = shallow(<Preview {...orderData} />);
  expect(wrapper).toMatchSnapshot();
});
