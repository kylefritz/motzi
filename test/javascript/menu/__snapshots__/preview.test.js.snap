// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview snapshot 1`] = `
<Fragment>
  <Subscription
    deadlineDay="Sunday"
    user={
      Object {
        "breadsPerWeek": 1,
        "credits": 9,
        "email": "kyle.p.fritz@gmail.com",
        "hashid": "Dot9gKn9w",
        "id": 584273342,
        "name": "Kyle Fritz",
      }
    }
  />
  <h2
    id="menu-name"
  >
    week 5
  </h2>
  <Deadline
    menu={
      Object {
        "createdAt": "2019-11-02T14:01:23.820-04:00",
        "deadline": "2019-11-03T23:59:59.000-04:00",
        "deadlineDay": "Sunday",
        "id": 921507399,
        "isCurrent": true,
        "items": Array [
          Object {
            "day1": true,
            "day2": true,
            "description": "",
            "id": 3,
            "image": "bread-baguette.jpg",
            "marketplaceOnly": false,
            "name": "Baguette",
            "price": 3,
            "subscriberOnly": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "Mix of modern wheats and ancient Einkorn for the best of both worlds.",
            "id": 1,
            "image": "bread2-002.webp",
            "marketplaceOnly": false,
            "name": "Classic",
            "price": 4,
            "subscriberOnly": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "ony subscribers can get cookies",
            "id": 2,
            "marketplaceOnly": false,
            "name": "Cookies",
            "price": 4,
            "subscriberOnly": true,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "too small for subscribers",
            "id": 4,
            "marketplaceOnly": true,
            "name": "Marketplace only item",
            "price": 2,
            "subscriberOnly": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "too small for subscribers",
            "id": 5,
            "marketplaceOnly": true,
            "name": "Another small item",
            "price": 1.5,
            "subscriberOnly": false,
          },
          Object {
            "day1": true,
            "day2": true,
            "description": "Skip this week.",
            "id": 0,
            "name": "Skip",
            "subscriberOnly": false,
          },
        ],
        "name": "week 5",
        "subscriberNote": "Orci varius",
      }
    }
  />
  <BakersNote
    note="Orci varius"
  />
  <h5>
    Items
  </h5>
  <Items
    items={
      Array [
        Object {
          "day1": true,
          "day2": true,
          "description": "",
          "id": 3,
          "image": "bread-baguette.jpg",
          "marketplaceOnly": false,
          "name": "Baguette",
          "price": 3,
          "subscriberOnly": false,
        },
        Object {
          "day1": true,
          "day2": true,
          "description": "Mix of modern wheats and ancient Einkorn for the best of both worlds.",
          "id": 1,
          "image": "bread2-002.webp",
          "marketplaceOnly": false,
          "name": "Classic",
          "price": 4,
          "subscriberOnly": false,
        },
        Object {
          "day1": true,
          "day2": true,
          "description": "ony subscribers can get cookies",
          "id": 2,
          "marketplaceOnly": false,
          "name": "Cookies",
          "price": 4,
          "subscriberOnly": true,
        },
        Object {
          "day1": true,
          "day2": true,
          "description": "too small for subscribers",
          "id": 4,
          "marketplaceOnly": true,
          "name": "Marketplace only item",
          "price": 2,
          "subscriberOnly": false,
        },
        Object {
          "day1": true,
          "day2": true,
          "description": "too small for subscribers",
          "id": 5,
          "marketplaceOnly": true,
          "name": "Another small item",
          "price": 1.5,
          "subscriberOnly": false,
        },
        Object {
          "day1": true,
          "day2": true,
          "description": "Skip this week.",
          "id": 0,
          "name": "Skip",
          "subscriberOnly": false,
        },
      ]
    }
  />
</Fragment>
`;
